Add command to change timeout in RfReceive

DarkPh0eNixKrg opened the pull request:
## Description:

Add command to change timeout in RfReceive
My door bell transmiter (Quhwa/1byone) sending ~70 payload (protocol 22).
I make rule to send mqtt status, but mqtt send 2-3 times payload.
To send 1 mqtt payload i was change "RF_TIME_AVOID_DUPLICATE" and add command to can change it.
But i dont know how to save timeout. After reboot timeout change to default.

I newbee in Tasmota (((
Sorry for my English

## Checklist:
  - [x] The pull request is done against the latest development branch
  - [x] Only relevant files were touched
  - [x] Only one feature/fix was added per PR and the code change compiles without warnings
  - [x] The code change is tested and works with Tasmota core ESP8266 V.2.7.4.9
  - [x] The code change is tested and works with Tasmota core ESP32 V.2.0.2.3
  - [x] I accept the [CLA](https://github.com/arendst/Tasmota/blob/development/CONTRIBUTING.md#contributor-license-agreement-cla).

DarkPh0eNixKrg committed:
Add command to change timeout in RfReceive

Add command to change timeout in RfReceive
My door bell transmiter (Quhwa/1byone) sending ~70 payload (protocol 22).
I make rule to send mqtt status, but mqtt send 2-3 times payload.
To send 1 mqtt payload i was change "RF_TIME_AVOID_DUPLICATE" and add command to can change it.
But i dont know how to save timeout. After reboot timeout change to default.

I newbee in Tasmota (((
Sorry for my English

barbudor commented on tasmota/xdrv_17_rcswitch.ino:
```diff
@@ -1,18 +1,14 @@
 /*
   xdrv_17_rcswitch.ino - RF transceiver using RcSwitch library for Tasmota
-
```
Do not delete lines from the header
This is the standard Tasmota header.
Please set them back

DarkPh0eNixKrg replied:
Corrected

barbudor commented on tasmota/xdrv_17_rcswitch.ino:
```diff
@@ -27,24 +23,26 @@
 
-#define RF_TIME_AVOID_DUPLICATE 1000  // Milliseconds
+int RF_TIME_AVOID_DUPLICATE = 3000;  // Use variable to change "RfTimeOut" to get new data from RfReceive
+//#define RF_TIME_AVOID_DUPLICATE 1000  // Milliseconds
```
You can remove that line

DarkPh0eNixKrg replied:
Corrected

barbudor commented on tasmota/xdrv_17_rcswitch.ino:
```diff
@@ -27,24 +23,26 @@
 RCSwitch mySwitch = RCSwitch();
 
-#define RF_TIME_AVOID_DUPLICATE 1000  // Milliseconds
+int RF_TIME_AVOID_DUPLICATE = 3000;  // Use variable to change "RfTimeOut" to get new data from RfReceive
```
Do not change the default value.
3000 may be the one that suits your specific use-case but in many other case this will break user's expectation as it will prevent users to send the same command within a short time. For exemple for a dimmer or a sound volume you need to press the key multiple time. Waiting 3 seconds between each press is not acceptable.

Also do not keep the name as fully capitalized. `RfTimeToAvoidDupplicate` would be better for a variable. Captalized should be only for preprocessors macros/constants.

DarkPh0eNixKrg replied:
Corrected

barbudor reviewed (commented):
You can set the value at each reboot using a boot rule such as `ON System#Boot DO RFTImeout 3000 ENDON`

DarkPh0eNixKrg committed:
Change some line for pull request

RF_TIME_AVOID_DUPLICATE → RfTimeToAvoidDupplicate
int RfTimeToAvoidDupplicate = 3000; → int RfTimeToAvoidDupplicate = 1000;

Jason2866 commented:
Please remove formatting style changes.

DarkPh0eNixKrg commented:
> c

I`m newbee, pls show or write what is formatting style need remove.

Jason2866 committed:
remove whitespaces

arendst commented:
I've seen your effort and will make a more permament solution. Hold on...

arendst commented:
Implemented my way. Thx.
